### PR TITLE
Makes math helpers defines for performance

### PR DESCRIPTION
**Pull request by kevinz000**
What it says on the tin!

### CODE REVIEW

File: code/__defines/math.dm [hunk 1]
```diff
@@ -1,18 +1,228 @@
...
+// Secant
+#define SEC(x) (1 / cos(x))
+
+// Cosecant
+#define CSC(x) (1 / sin(x))
+
+#define ATAN2(x, y) ( !(x) && !(y) ? 0 : (y) >= 0 ? arccos((x) / sqrt((x)*(x) + (y)*(y))) : -arccos((x) / sqrt((x)*(x) + (y)*(y))) )
+
+// Greatest Common Divisor - Euclid's algorithm
+/proc/GCD(a, b)
```
Comment (Atermonera) [inline code/__defines/math.dm]: Why are you moving helper functions out of a file for helper functions and into a file for defines?

File: code/modules/mob/living/carbon/metroid/powers.dm [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/mob/living/carbon/slime/proc/Wrap(var/mob/living/M) // This is a proc for the clicks
+/mob/living/carbon/slime/proc/WRAP(var/mob/living/M) // This is a proc for the clicks
```
Comment (Atermonera) [inline code/modules/mob/living/carbon/metroid/powers.dm]: This isn't a define, so it shouldn't be all caps

Comment (kevinz000) [reply code/modules/mob/living/carbon/metroid/powers.dm]: Oops.

File: code/__defines/math.dm (hunk 1 above)
Comment (kevinz000) [reply code/__defines/math.dm]: I consolidated all math based stuff into a single math file, could put it back in a math.dm under helpers I guess?

File: code/__defines/math.dm [hunk 3]
```diff
@@ -1,18 +1,228 @@
...
+// Cotangent
+#define COT(x) (1 / TAN(x))
+
+// Secant
+#define SEC(x) (1 / cos(x))
+
+// Cosecant
+#define CSC(x) (1 / sin(x))
+
+#define ATAN2(x, y) ( !(x) && !(y) ? 0 : (y) >= 0 ? arccos((x) / sqrt((x)*(x) + (y)*(y))) : -arccos((x) / sqrt((x)*(x) + (y)*(y))) )
```
[the remark this answers is not in the document]
Comment (kevinz000) [reply code/__defines/math.dm]: This is the old Atan2 but as a define. 
/proc/Atan2(x, y)
	if(!x && !y) return 0
	var/a = arccos(x / sqrt(x*x + y*y))
	return y >= 0 ? a : -a
#define ATAN2(x, y) (
 !(x) && !(y) ? 0 : 
(y) >= 0 ? arccos((x) / sqrt((x)*(x) + (y)*(y))) :
 -arccos((x) / sqrt((x)*(x) + (y)*(y))) )

**Pull request merged**